packages/xstate-store/src/types.ts

```diff
@@ -8,7 +8,11 @@ export type Recipe<T, TReturn> = (state: T) => TReturn;
 
 export type EnqueueObject<TEmittedEvent extends EventObject> = {
   emit: {
-    [E in TEmittedEvent as E['type']]: (payload: Omit<E, 'type'>) => void;
+    [E in TEmittedEvent as E['type']]: IsEmptyObject<
+      Omit<E, 'type'>
+    > extends true
+      ? (_?: {}) => void
```

Comment:
this is wrong because you now permit strings, numbers and whatnot here - remember that `{}` means in TS basically `NonNullable<unknown>`, it **doesn't** mean "an empty object"